src/Data/Tree.purs

```diff
@@ -60,14 +59,15 @@ instance functorTree :: Functor Tree where
   map f (Tree a ts) = Tree (f a) (map f <$> ts)
 
 instance applyTree :: Apply Tree where
-  apply (Tree f tf) t@(Tree a ta) = Tree (f a) (map (f <$>) ta <> map (<*> t) tf)
+  apply (Tree f tf) t@(Tree a ta) = Tree (f a) (map (map f) ta <> map (applyFlipped t) tf)
+    where applyFlipped x y = apply y x
```

Comment:
Updated now - thanks @LiamGoodacre for pointing out the use of operator sections